tensorflow/python/ops/nn_test.py

```diff
@@ -303,6 +303,19 @@ def testL2NormalizeGradient(self):
       print("L2Normalize gradient err = %g " % err)
       self.assertLess(err, 1e-4)
 
+  @test_util.run_in_graph_and_eager_modes
+  def testL2NormalizeComplex(self):
+    x_shape = [20, 7, 3]
+    for dtype in [np.complex64, np.complex128]:
+      np.random.seed(1)
+      x_np = np.random.random_sample(x_shape).astype(dtype)
+      x_np = x_np + x_np * 1j
```

Comment:
Awesome, thanks for adding this feature so quickly! One thing I wonder about is it looks like the unit test uses the same real and imaginary components for all elements? Concerned this could hypothetically let through bugs related to how the real/imag components are handled separately in the implementation. Could instead use something like 
```
x_np = (np.random.random_sample(x_shape).astype(dtype) +
        np.random.random_sample(x_shape).astype(dtype) * 1j)
```
?

Reply:
Updated.